src/ripple/app/tx/impl/OfferStream.cpp

```diff
@@ -225,6 +226,96 @@ TOfferStreamBase<TIn, TOut>::step()
             continue;
         }
 
+        bool const inIsXRP = isXRP(offer_.issueIn());
+        bool const outIsXRP = isXRP(offer_.issueOut());
+        if (view_.rules().enabled(fixRmSmallIncreasedQOffers) &&
+            (inIsXRP || outIsXRP))
+        {
+            // If an offer is small (defined by the number of drops available),
+            // and the effective quality is different from its initial quality,
+            // remove it. Such offers may block order books.
+            auto const effectiveAmounts = [&] {
+                if (offer_.owner() != offer_.issueOut().account &&
+                    *ownerFunds_ < offer_.amount().out)
+                {
+                    // adjust the amounts by owner funds
+                    return offer_.quality().ceil_out(
+                        offer_.amount(), *ownerFunds_);
+                }
+                return offer_.amount();
+            }();
+
+            XRPAmount const thresholdToRm{2};
```

Comment:
A comment to explain why this is 2 rather than 1 would be helpful in understanding the issue.

Reply:
@donovanhide I changed it back to 1. Anything with values larger than 1 should give a quality that's reasonably in line with the initial quality. The reason I made it larger is I was thinking "removing some more tiny offers is a good thing", but they should be removed with regular operation. There's no need for the larger threshold.

Fixed in the patch `7474063b72 [fold] Change the threshold and minor refactor`